src/test/java/com/scottlogic/filters/combine/AndCombineFiltersTest.java

```diff
@@ -0,0 +1,128 @@
+package com.scottlogic.filters.combine;
+
+import com.scottlogic.UserPost;
+import com.scottlogic.filters.AuthorPostFilter;
+import com.scottlogic.filters.DatePostFilter;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class AndCombineFiltersTest {
+    UserPost userPostJane = new UserPost("Jane Smith",
+            OffsetDateTime.of(2020, 1, 3, 7, 12, 3, 0, ZoneOffset.UTC),
+            "Hello World!", 2);
+
+    UserPost userPostJoe1 = new UserPost("Joe Bloggs",
+            OffsetDateTime.of(2020, 1, 3, 8, 53, 34, 0, ZoneOffset.UTC),
+            "Another example post.", 1);
+
+    UserPost userPostJoe2 = new UserPost("Joe Bloggs",
+            OffsetDateTime.of(2020, 3, 12, 13, 22, 12, 0, ZoneOffset.UTC),
+            "An example of a post \nwith lines breaks.", 3);
+
+
+    @Mock
+    AuthorPostFilter localAuthorPostFilter = mock(AuthorPostFilter.class);
+    @Mock
+    DatePostFilter localDatePostFilter = mock(DatePostFilter.class);
+
+
+
+    @Test
+    void filter_null_null() {
+        when(localAuthorPostFilter.filter(null)).thenReturn(null);
+        when(localDatePostFilter.filter(null)).thenReturn(null);
+
+        List<UserPost> actual = new AndCombineFilters(localAuthorPostFilter, localDatePostFilter).filter(null);
+
+        assertNull(actual);
+    }
+
+    @Test
+    void filter_EmptyList_EmptyList() {
+        List<UserPost> inputList = Collections.emptyList();
+        List<UserPost> expected = Collections.emptyList();
+
+        when(localAuthorPostFilter.filter(inputList)).thenReturn(List.of());
+
+        when(localDatePostFilter.filter(List.of())).thenReturn(List.of());
+
+        List<UserPost> actual = new AndCombineFilters(localAuthorPostFilter, localDatePostFilter).filter(inputList);
+
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void filter_1ItemListMatchBoth_1ItemList() {
+        List<UserPost> inputList = List.of(userPostJane);
+        List<UserPost> expected = List.of(userPostJane);
+
+        when(localAuthorPostFilter.filter(inputList)).thenReturn(List.of(userPostJane));
+        when(localDatePostFilter.filter(List.of(userPostJane))).thenReturn(List.of(userPostJane));
+
+        List<UserPost> actual = new AndCombineFilters(localAuthorPostFilter, localDatePostFilter).filter(inputList);
+
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void filter_1Item1Match_EmptyList() {
+        List<UserPost> inputList = List.of(userPostJane);
+        List<UserPost> expected = List.of();
+
+        when(localAuthorPostFilter.filter(inputList)).thenReturn(List.of(userPostJane));
+        when(localDatePostFilter.filter(List.of(userPostJane))).thenReturn(List.of());
```

Comment:
A minor point at the moment your `mocks` are relying on the implementation detail that the first filter given is applied first. This isn't technically a requirement and so ideally the tests should be agnostic to this detail. Might not necessarily be worth the extra effort to do it on all of the tests as some of them can get quite verbose doing it that way but maybe worth doing it on a couple of the smaller ones to get used to the idea.